### PR TITLE
Bug 1446977 - try to fix NaNs causing panics

### PR DESCRIPTION
**Pull request by djmitche**
I don't know if this will fix the issue, but it removes a bunch of buggy stuff that might be responsible, and logs a little more clearly about the issue.

### CODE REVIEW

**Comment by djmitche** [comment]
The ffjson issue is https://github.com/pquerna/ffjson/issues/241

It seems that it will refuse to Marshal NaN, since that's the error causing the panic.

File: cmd/statsum/main.go [hunk 1]
```diff
@@ -111,9 +112,9 @@ func main() {
 			fmt.Println("Error: ", err)
 			os.Exit(1)
 		}
-		json := args["--json"].(bool)
+		isJson := args["--json"].(bool)
```
Comment (petemoore) [inline cmd/statsum/main.go]: nice

File: server/server_test.go [hunk 2]
```diff
@@ -105,6 +106,16 @@ func TestHandleCorrectPrefix(t *testing.T) {
...
 
+func TestHandleInvalidNaN(t *testing.T) {
+	// this is *invalid* JSON -- NaN is not a legal identifier in a JSON document
+	var testBody = []byte("{\"counters\":null,\"measures\":[{\"k\":\"test-measure\",\"v\":[NaN]}]}")
+	r, _ := http.NewRequest("POST", "https://statsum.local/v1/project/prefix", bytes.NewReader(testBody))
+	r.Header.Set("Content-Type", "application/json")
+	r.Header.Set("Authorization", auth("prefix"))
+	w, _ := doTestRequest(r, nil)
+	assert(w.Code != http.StatusOK, "did not fail with invalid JSON")
+}
```
Comment (petemoore) [inline server/server_test.go]: Nice test.

File: server/signalfx.go [hunk 3]
```diff
@@ -35,6 +36,10 @@ func (s *signalfxDrain) Name() string {
 }
 
 func (s *signalfxDrain) Send(name string, value float64, time time.Time) {
+	if math.IsNaN(value) {
+		fmt.Printf("Ignoring NaN value for metric %s\n", name)
+		return
+	}
```
Comment (petemoore) [inline server/signalfx.go]: Seems sensible.

**Comment by petemoore** [comment]
@djmitche I wonder why taskcluster-github didn't update this PR with a CI task group... 🤔

**Comment by djmitche** [comment]
Because `.taskcluster.yml` only specifies to run on pushes

**Comment by petemoore** [comment]
> Because `.taskcluster.yml` only specifies to run on pushes

Created https://github.com/taskcluster/statsum/pull/14 - although would mean s3 bucket write access for untrusted PRs - so maybe we shouldn't...

**Pull request merged**